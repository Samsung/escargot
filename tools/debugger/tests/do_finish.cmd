b do_finish.js:24
c
finish
f
b do_finish.js:41
f
f
f
