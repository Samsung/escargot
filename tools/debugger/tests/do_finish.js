/*
 * Copyright (c) 2020-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

var a = 1;

function f() {
  with ({ a:4 }) {
    a = 2;
  }
  a = 3;
}

function g() {
  try {
    f();
    a = 4;
  } catch (e) {
  }
  a = 5;
}
g();
a = 6;

function* gen() {
   a = 1;
   yield 4;
}

gen().next();
a = 2;
