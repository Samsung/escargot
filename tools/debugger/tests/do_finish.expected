Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_finish.js:20
(escargot-debugger) b do_finish.js:24
Breakpoint 1 at tools/debugger/tests/do_finish.js:24 (in f() at line:22, col:1)
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_finish.js:24 (in f() at line:22, col:1)
(escargot-debugger) finish
Stopped at tools/debugger/tests/do_finish.js:32 (in g() at line:29, col:1)
(escargot-debugger) f
Stopped at tools/debugger/tests/do_finish.js:38
(escargot-debugger) b do_finish.js:41
Breakpoint 2 at tools/debugger/tests/do_finish.js:41 (in gen() at line:40, col:1)
(escargot-debugger) f
Stopped at breakpoint:2 tools/debugger/tests/do_finish.js:41 (in gen() at line:40, col:1)
(escargot-debugger) f
Stopped at tools/debugger/tests/do_finish.js:46
(escargot-debugger) f
Connection closed.
